src/server/lib/validateUrl.ts

```diff
@@ -27,15 +31,18 @@ export const validateReturnUrl = (returnUrl = ''): string => {
 			throw 'Invalid hostname';
 		}
 
-		// if the hostname is on the profile subdomain, we also want to check the pathname is valid
-		// and keep any query params
-		if (url.hostname.startsWith('profile.')) {
-			// then check the pathname is valid
-			if (invalidPaths.some((path) => url.pathname.startsWith(path))) {
-				throw 'Invalid pathname';
+		// if valid subdomains are present, we can return the url with the query params
+		if (
+			validSubdomains.some((subdomain) => url.hostname.startsWith(subdomain))
```

Comment:
If I have a domain that is `support-this-baddy.theguardian.com` or `support-this-naughty.oops.com` - would this still work?

Reply:
It would not, we're specifically looking for the `support.` string at the start of the hostname and `.theguardian.com`

Reply:
Ah, [this was the code I was missing](https://github.com/guardian/gateway/pull/2781/files?diff=split&w=0#diff-ca394c2b701ae7115acc7166b12451ced999603cb59d6a1a0691c8eb6e9a7790L26-L28).